platform-sdk/swirlds-state-impl/src/main/java/com/swirlds/state/merkle/MerkleStateRoot.java

```diff
@@ -293,652 +131,199 @@ public T copy() {
-                throw new IllegalStateException("Unexpected type for k/v state " + stateKey);
-            }
-        }
-
-        @NonNull
-        @Override
-        public <T> ReadableSingletonState<T> getSingleton(@NonNull String stateKey) {
-            final ReadableSingletonState<T> instance = (ReadableSingletonState<T>) singletonInstances.get(stateKey);
-            if (instance != null) {
-                return instance;
-            }
-
-            final var md = stateMetadata.get(stateKey);
-            if (md == null || !md.stateDefinition().singleton()) {
-                throw new IllegalArgumentException("Unknown singleton state key '" + stateKey + "'");
-            }
+    public abstract long getCurrentRound();
 
-            final var node = findNode(md);
-            if (node instanceof SingletonNode s) {
-                final var ret = createReadableSingletonState(md, s);
-                singletonInstances.put(stateKey, ret);
-                return ret;
-            } else {
-                // This exception should never be thrown. Only if "findNode" found the wrong node!
-                throw new IllegalStateException("Unexpected type for singleton state " + stateKey);
-            }
-        }
 
-        @NonNull
-        @Override
-        public <E> ReadableQueueState<E> getQueue(@NonNull String stateKey) {
-            final ReadableQueueState<E> instance = (ReadableQueueState<E>) queueInstances.get(stateKey);
-            if (instance != null) {
-                return instance;
-            }
+    // Migration things
 
-            final var md = stateMetadata.get(stateKey);
-            if (md == null || !md.stateDefinition().queue()) {
-                throw new IllegalArgumentException("Unknown queue state key '" + stateKey + "'");
-            }
+    // TODO: decide on method from VirtualMapMigration
+    @Override
+    public MerkleNode migrate(@NonNull final Configuration configuration, int version) {
+        if (version < CURRENT_VERSION) {
+            AtomicLong totalMigratedObjects = new AtomicLong(0);
+            AtomicLong totalMigrationTimeMs = new AtomicLong(0);
+            AtomicLong totalValidationTimeMs = new AtomicLong(0);
+
+            // Config constants
+            // Threads which iterate over the given Virtual Map, perform some operation and write into its own output queue/buffer
+            final int THREAD_COUNT = 1;
+            final long MEGA_MAP_MAX_KEYS_HINT = 1_000_000_000;
+            final boolean VALIDATE_MIGRATION_FF = true;
+
+            final MerkleDbConfig merkleDbConfig = configuration.getConfigData(MerkleDbConfig.class);
+            final var tableConfig = new MerkleDbTableConfig(
+                    (short) 1,
+                    DigestType.SHA_384,
+                    MEGA_MAP_MAX_KEYS_HINT,
+                    merkleDbConfig.hashesRamToDiskThreshold());
+            final var virtualMapLabel = "VirtualMap";
```

Comment:
Maybe "StateMap" or "RootMap"?

Reply:
I favor the "StateMap," which indicates that this map holds all state data.

@imalygin wdyt?